Data/BankerQueue/LazyLists.idr

```diff
@@ -0,0 +1,189 @@
+||| This is not actually a general lazy list library. It's really
+||| intended for implementing banker's queues. It also has some
+||| junk about regular lists for that purpose.
+module Data.BankerQueue.LazyLists
+
+%default total
+%access public
+
+||| A lazy-spined list.
+data LList : Type -> Type where
```

Comment:
I am wondering whether it could be interesting to use something like a CoList (https://github.com/idris-lang/Idris-dev/blob/0c953a9c2d4e2f1fb11b5022055b60af4ee45f19/libs/contrib/Data/CoList.idr) instead in this case.